### PR TITLE
Add sections for class, component and type.

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

Add initial sections in the documentation for the new `class`, `component` and `type` attributes, that the PR https://github.com/netdata/netdata/pull/10961 adds to the stock alarms configuration files.

##### Component Name

Health

##### Test Plan

Not needed.

##### Additional Information

The documentation can be perhaps extended to provide information on how those attributes are used in the cloud.

### CODE REVIEW

File: health/REFERENCE.md
Comment (ilyam8) [inline health/REFERENCE.md:142]: What about allowed characters? Are there any restrictions? Are spaces allowed?

Comment (thiagoftsm) [reply health/REFERENCE.md:142]: Considering the current code in https://github.com/netdata/netdata/pull/10961, I think we would have the same restrictions that JSON has, except for quotes that we are stripping, @MrZammler can you confirm this?

File: health/REFERENCE.md
Comment (thiagoftsm) [inline health/REFERENCE.md:148]: Can `component` and `class` to have the exact same values? If it is not possible, we need to update the PR #10961.

File: health/REFERENCE.md
Comment (thiagoftsm) [inline health/REFERENCE.md:159]: Right now our current code gives a big freedom for users, I do not see this as a problem, on the other hand this freedom also brings the necessity to start thinking on binary trees to get data related to these possible values, I did not see them in the previous PR.

File: health/REFERENCE.md
Comment (MrZammler) [reply health/REFERENCE.md:142]: Hi Thiago, yes, I will do some tests and report back.

Comment (thiagoftsm) [reply health/REFERENCE.md:142]: I will wait for your tests before to proceed with the review, the question @ilyam8 raised is very important for that PR.
Please, also take a look in the next question I wrote, because it also impacts the PR.

Comment (MrZammler) [reply health/REFERENCE.md:142]: The final values of these strings are consumed by the cloud. I think they're used to group together alarms. That's why I've added @dim08 (the docs should reflect that too - that's their main use after all). 

So I think if the user chooses to have a `class` value of `System 1 - my great machine!!^@&#` to some alarms, and the cloud is able to consume it, we should allow it to be read and pass through the agent. We do strip quotes (both single and double), so this could go in the docs maybe? (But again if grouping is the target, then it should work).

The same could be true for the info field? I think any character that can be passed through JSON should be allowed, unless the cloud has other restrictions that I'm not aware of. Of course some kind of non-utf valid character, will likely break things, so if it's needed we should sanitize the fields (and the info one too).

File: health/REFERENCE.md
Comment (MrZammler) [reply health/REFERENCE.md:148]: I don't recall that being a requirement... or even discussed at all for that matter. I think a requirement like that would have been explicitly stated.

Comment (ilyam8) [reply health/REFERENCE.md:148]: > Can component and class to have the exact same values?

Yes, "Unknown" 😄

File: health/REFERENCE.md
Comment (thiagoftsm) [reply health/REFERENCE.md:159]: According your previews [answer](https://github.com/netdata/netdata/pull/10984#discussion_r614888815) this is liked to be used on cloud only, so I will mark this as resolved, but case we start using this on agent, it will be necessary to add more control on agent side.

File: health/REFERENCE.md
Comment (thiagoftsm) [reply health/REFERENCE.md:142]: > What about allowed characters? Are there any restrictions? Are spaces allowed?

I do not know exactly what the product team wants, but I made some tests on Friday and today, and I saw that we can use any character for these events, on the other hand I discovered a bug thanks these tests, but I will bring a complete different PR that will address it.

Please, talk with the cloud team to confirm they are expecting any ASCII character.

**Review by joelhans — COMMENTED** [review]
No specific changes for me to request at the moment, however I'm wondering if it would be helpful to list the `class`, `component`, and `type` values that are already being used by our preconfigured alarms. For example, the `class` section might have a paragraph like:

> If you want to use an existing class already defined in Netdata's preconfigured alarms, use one of the following: `Database`, `Web`, `RAM`, ...

**Comment by MrZammler** [comment]
> No specific changes for me to request at the moment, however I'm wondering if it would be helpful to list the `class`, `component`, and `type` values that are already being used by our preconfigured alarms. For example, the `class` section might have a paragraph like:
> 
> > If you want to use an existing class already defined in Netdata's preconfigured alarms, use one of the following: `Database`, `Web`, `RAM`, ...

Hi Joel! I've added a table of all the stock `class` attributes we use. Not sure if we want it like this (a table), have a description or not of each attribute?

The `component` and `type` attributes can get really long for a table though, and I don't know if it makes sense to describe them all ?

**Comment by OdysLam** [comment]
What's the status of this PR? @MrZammler 

Let's try to merge this since the relevant code is already merged (I believe).

**Comment by MrZammler** [comment]
Well, considering it's my first document-related PR, I'm ok if you guys are too. Yes, the code is already commited (along with the additions of those fields in the alarms configuration).

File: health/REFERENCE.md
Comment (thiagoftsm) [reply health/REFERENCE.md:142]: @MrZammler the unique doubt I have to approve the PR is this discussion, do we arrive in the conclusion that cloud is allowing any ASCII character?

Comment (MrZammler) [reply health/REFERENCE.md:142]: Hi Thiago, I'm looking into this (the characters), thanks.

File: health/REFERENCE.md
Comment (ilyam8) [inline health/REFERENCE.md:171]: A lot of classes have no _Groups alarms for_ descriptions. 

- Are we done with `class`ification? if not perhaps it makes sense to not add this table? @joelhans 

- And what's Other? do we use it?

- And i think we need to mentions that if not set we use `Unknown` for all the new attributes.

File: health/REFERENCE.md
Comment (MrZammler) [reply health/REFERENCE.md:142]: Sorry Thiago for being late to address this.... 

So, I did a test on the info field (I can't find the class/component/type strings on the cloud...), and the cloud will display an info string with e.g. `number !@#$%^&*()_+=-}{][;:?/><,. of metrics lost ....`. I've asked on cloud-ops, will update.

File: health/REFERENCE.md
Comment (MrZammler) [reply health/REFERENCE.md:171]: Hi Ilya. Yes, we do have some `Other` as class in some alerts. I will add the `Unknown` case in the docs, thanks.

Comment (MrZammler) [reply health/REFERENCE.md:170]: I will add the rest of the descriptions, and go from there with Joel if needed. I suppose it makes sense to keep the table.

Comment (ilyam8) [reply health/REFERENCE.md:170]: Ok, just an idea - consider using _collapsible section_ feature for the tables to keep the doc tidy. Could be just me, feel free to ignore the suggestion if you don't like it.

<details>
<summary>default class attributes</summary>


| Class                    | Groups alarms for:                              |
| ------------------------ | ----------------------------------------------- |
| Ad Filtering             | Services related to Ad Filtering (like pi-hole) |
| Certificates             | 
| Cgroups                  |
| Computing                | 
| Containers               |
| Database                 | Database systems (e.g. MySQL, Postgress, etc)   |
| Data Sharing             |
| DHCP                     |
| DNS                      | 
| Kubernetes               |
| KV Storage               |
| Linux                    | Services specific to Linux (e.g. systemd)       |
| Messaging                |
| Netdata                  | Internal Netdata components monitoring          |
| Other                    |
| Power Supply             |
| Search engine            |
| Storage                  |
| System                   | General system alarms (e.g. cpu, network, etc.) |
| Virtual Machine          | Virtual Machine software                        |
| Web Proxy                |
| Web Server               | Web server software (e.g. Apache, ngnix, etc.)  |
| Windows                  |                                                 |

</details>

Comment (MrZammler) [reply health/REFERENCE.md]: Whoa! Neat! Added!

**Comment by MrZammler** [comment]
So I think we're missing an ok or changes from @joelhans and an ok from cloud on the characters.

File: health/REFERENCE.md
Comment (ilyam8) [inline health/REFERENCE.md:145]:
```suggestion
<summary>Netdata's stock alarms use the following `class` attributes by default, but feel free to adjust for your own requirements.</summary>
```

Comment (ilyam8) [reply health/REFERENCE.md:145]: i am wrong, nvm 😄

File: health/REFERENCE.md
Comment (joelhans) [inline health/REFERENCE.md:175]:
```suggestion
If an alarm configuration is missing the `class` line, its value will default to `Unknown`.
```

File: health/REFERENCE.md
Comment (joelhans) [inline health/REFERENCE.md:184]:
```suggestion
As with the `class` line, if `component` is missing from the configuration, its value will default to `Unknown`.
```

File: health/REFERENCE.md
Comment (joelhans) [inline health/REFERENCE.md:188]:
```suggestion
This indicates the type of error (or general problem area) that the alarm or template applies to. For example, `Latency` can be used for alarms that trigger on latency issues in network interfaces, web servers, or database systems. Example:
```

**Review by joelhans — COMMENTED** [review]
Just a few small tweaks. Overall it looks good.

**Review by joelhans — APPROVED** [review]

**Review by ilyam8 — APPROVED** [review]

File: health/REFERENCE.md
Comment (MrZammler) [reply health/REFERENCE.md:142]: Furthermore, I've received an alert email from the cloud, after having set a class attribute to ` Classification: This is a class !@#$%^*()_&+=-}{][:;?><,./ ` and it correctly appeared in the email message:

**Review by thiagoftsm — APPROVED** [review]
After last explanation from @MrZammler , LGTM!

**Pull request merged**
